File: conteo_electoral_app/lib/models/candidato.dart
class Candidato {
  final int id;
  final String nombre;
  final String apellido;
  final int? partidoId;
  final String partidoNombre;
  final String? partidoSigla;
  final int? cargoId;
  final String? cargoNombre;
  final int? listaId;
  final String? listaNombre;
  final int? numeroLista;
  final String? fotoUrl;
  final int? eleccionesId;
  final int? ordenEnLista;
  final String? tipo;
  final bool? principal;
  final bool? activo;

  Candidato({
    required this.id,
    required this.nombre,
    required this.apellido,
    this.partidoId,
    required this.partidoNombre,
    this.partidoSigla,
    this.cargoId,
    this.cargoNombre,
    this.listaId,
    this.listaNombre,
    this.numeroLista,
    this.fotoUrl,
    this.eleccionesId,
    this.ordenEnLista,
    this.tipo,
    this.principal,
    this.activo,
  });

  factory Candidato.fromJson(Map<String, dynamic> json) {
    return Candidato(
      id: json['id'],
      nombre: json['nombre'],
      apellido: json['apellido'],
      partidoId: json['partidoId'],
      partidoNombre: json['partidoNombre'] ?? 'Independiente',
      partidoSigla: json['partidoSigla'],
      cargoId: json['cargoId'],
      cargoNombre: json['cargoNombre'],
      listaId: json['listaId'],
      listaNombre: json['listaNombre'],
      numeroLista: json['numeroLista'],
      fotoUrl: json['fotoUrl'],
      eleccionesId: json['eleccionesId'],
      ordenEnLista: json['ordenEnLista'],
      tipo: json['tipo'],
      principal: json['principal'],
      activo: json['activo'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'apellido': apellido,
      'partidoId': partidoId,
      'partidoNombre': partidoNombre,
      'partidoSigla': partidoSigla,
      'cargoId': cargoId,
      'cargoNombre': cargoNombre,
      'listaId': listaId,
      'listaNombre': listaNombre,
      'numeroLista': numeroLista,
      'fotoUrl': fotoUrl,
      'eleccionesId': eleccionesId,
      'ordenEnLista': ordenEnLista,
      'tipo': tipo,
      'principal': principal,
      'activo': activo,
    };
  }

  String get nombreCompleto => '$nombre $apellido';
}

File: backend/src/main/java/com/electoral/entities/Candidato.java
package com.electoral.entities;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "candidatos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Candidato {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(nullable = false, length = 100)
    private String apellido;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "partido_id")
    private Partido partido;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cargo_id")
    private Cargo cargo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lista_id")
    private ListaElectoral lista;

    @Column(name = "foto_url", length = 500)
    private String fotoUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "elecciones_id")
    private Eleccion elecciones;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "provincia_id")
    private Provincia provincia;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "canton_id")
    private Canton canton;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parroquia_id")
    private Parroquia parroquia;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tipo_circunscripcion_id")
    private TipoCircunscripcion tipoCircunscripcion;

    @Column(name = "posicion_lista")
    private Integer posicionLista;

    @Column(name = "orden_aparicion")
    private Integer ordenAparicion;

    @Column(name = "orden_en_lista")
    private Integer ordenEnLista;

    @Column(length = 20)
    @Builder.Default
    private String tipo = "PRINCIPAL";

    @Builder.Default
    @Column(nullable = false)
    private Boolean principal = true;

    @Builder.Default
    @Column(nullable = false)
    private Boolean activo = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    public String getNombreCompleto() {
        return nombre + " " + apellido;
    }
}